src/checkpoint/container/view_serialize.h

```diff
@@ -235,6 +245,88 @@ inline void serialize(
 #endif
 }
 
+template <typename SerializerT, typename T, typename... Args>
+inline void serialize_impl(SerializerT& s, Kokkos::DynRankView<T,Args...>& view) {
+  using ViewType = Kokkos::DynRankView<T,Args...>;
+  using ArrayLayoutType = typename ViewType::traits::array_layout;
+
+  // Serialize the label for the view which is used to construct a new view with
+  // the same label. Labels may not be unique and are for debugging Kokkos::View
+  auto const label = serializeViewLabel(s,view);
+
+  // Serialize the total number of runtime dimensions
+  unsigned dims = 0;
+  if (!s.isUnpacking()) {
+    dims = view.rank();
+  }
+  s | dims;
+
+  // Serialize the Kokkos layout data, including the extents, strides
+  ArrayLayoutType layout;
+
+  // Make sure we serialize all 7 dimensions, instead of just `dims`. The
+  // other dimensions contain a 1 to make sure the size comes out correctly
+  if (s.isUnpacking()) {
+    serializeLayout<SerializerT>(s, 7, layout);
+  } else {
+    ArrayLayoutType layout_cur = view.layout();
+    serializeLayout<SerializerT>(s, 7, layout_cur);
+  }
+
+  // Construct a view with the layout and use operator= to propagate out
+  if (s.isUnpacking()) {
+    if (dims == 1) {
+      view = constructRankedView<ViewType,1>(label, std::make_tuple(layout));
+    } else if (dims == 2) {
+      view = constructRankedView<ViewType,2>(label, std::make_tuple(layout));
+    } else if (dims == 3) {
+      view = constructRankedView<ViewType,3>(label, std::make_tuple(layout));
+    } else if (dims == 4) {
+      view = constructRankedView<ViewType,4>(label, std::make_tuple(layout));
+    } else if (dims == 5) {
+      view = constructRankedView<ViewType,5>(label, std::make_tuple(layout));
+    } else if (dims == 6) {
+      view = constructRankedView<ViewType,6>(label, std::make_tuple(layout));
+    } else if (dims == 7) {
+      view = constructRankedView<ViewType,7>(label, std::make_tuple(layout));
+    } else {
+      checkpointAssert(
+        false,
+        "Serializing Kokkos::DynRankView is only supported up to 7 dimensions"
+      );
+    }
+  }
+
+  // Serialize the total number of elements in the Kokkos::View
+  size_t num_elms = view.size();
+  s | num_elms;
+
+  // Serialize whether the view is contiguous or not. Is this required?
```

Comment:
Why wouldn't this be required, unless we removed the `if (is_contig)` below and just used `TraverseManual` unconditionally? Does `DynRankView` guarantee that it's contiguous? Does it hurt to just check and do the detailed thing, in case that changes?